ledger/src/executor.rs

```diff
@@ -0,0 +1,421 @@
+    }
+
+    bank_utils::find_and_send_votes(
+        batch.sanitized_transactions(),
+        &tx_results,
+        replay_vote_sender,
+    );
+
+    let TransactionResults {
+        fee_collection_results,
+        execution_results,
+        rent_debits,
+        ..
+    } = tx_results;
+
+    if let Some(transaction_status_sender) = transaction_status_sender {
+        let transactions = batch.sanitized_transactions().to_vec();
+        let post_token_balances = if record_token_balances {
+            collect_token_balances(bank, batch, &mut mint_decimals)
+        } else {
+            vec![]
+        };
+
+        let token_balances =
+            TransactionTokenBalancesSet::new(pre_token_balances, post_token_balances);
+
+        transaction_status_sender.send_transaction_status_batch(
+            bank.clone(),
+            transactions,
+            execution_results,
+            balances,
+            token_balances,
+            rent_debits,
+            transaction_indexes.to_vec(),
+        );
+    }
+    let first_err = get_first_error(batch, fee_collection_results);
+    first_err.map(|(result, _)| result).unwrap_or(Ok(()))
+}
+
+// The dependency graph contains a set of indices < N that must be executed before index N.
+pub fn build_dependency_graph(
+    tx_account_locks_results: &[Result<TransactionAccountLocks>],
+) -> Result<Vec<HashSet<usize>>> {
+    if let Some(err) = tx_account_locks_results.iter().find(|r| r.is_err()) {
+        err.clone()?;
+    }
+    let transaction_locks: Vec<_> = tx_account_locks_results
+        .iter()
+        .map(|r| r.as_ref().unwrap())
+        .collect();
+
+    // build a map whose key is a pubkey + value is a sorted vector of all indices that
+    // lock that account
+    let mut indices_read_locking_account = HashMap::new();
+    let mut indicies_write_locking_account = HashMap::new();
+    transaction_locks
+        .iter()
+        .enumerate()
+        .for_each(|(idx, tx_account_locks)| {
+            for account in &tx_account_locks.readonly {
+                indices_read_locking_account
+                    .entry(**account)
+                    .and_modify(|indices: &mut Vec<usize>| indices.push(idx))
+                    .or_insert_with(|| vec![idx]);
+            }
+            for account in &tx_account_locks.writable {
+                indicies_write_locking_account
+                    .entry(**account)
+                    .and_modify(|indices: &mut Vec<usize>| indices.push(idx))
+                    .or_insert_with(|| vec![idx]);
+            }
+        });
+
+    Ok(PAR_THREAD_POOL.install(|| {
+        transaction_locks
+            .par_iter()
+            .enumerate()
+            .map(|(idx, account_locks)| {
+                // user measured value from mainnet; rarely see more than 30 conflicts or so
+                let mut dep_graph = HashSet::with_capacity(DEFAULT_CONFLICT_SET_SIZE);
+                let readlock_accs = account_locks.writable.iter();
```

Comment:
I get what's going on, but I think the naming of these readlock_accs and writelock_accs can be confusing at a glance. Still trying to come up with a better name though...